scripts/process_uploads.py

```diff
@@ -3,89 +3,63 @@
 Processes a new batch of samples from the UPLOAD bucket,
 into the MAIN and ARCHIVE buckets using the batch_move_files function. 
 
-Determines the files to be moved through a comparison of the 
-most recently updated CSV and the CSV uploaded in a previous batch run. 
+Assumes that the csv file has been processed in a previous step. 
 """
 
 import os
-import io
-import csv
-from os.path import basename, join
-from typing import List, Optional, Set, Tuple
+from os.path import join
+from typing import List, Optional, Tuple
 import click
 import hailtop.batch as hb
-from google.cloud import storage
-from joint_calling.upload_processor import batch_move_files, SampleGroup
-
 
-def samples_from_csv(bucket_name, prefix) -> Tuple[List[str], str]:
-    """Determines list of samples from a csv file within a specified bucket
+# Import SampleAPI
+from sample_metadata.api.sample_api import SampleApi
+from joint_calling.upload_processor import batch_move_files, SampleGroup
 
-    Assumptions
-    ==========
-    Only one .csv file exists in the upload bucket and previous batch directory.
-    """
 
-    client = storage.Client()
-    bucket = client.get_bucket(bucket_name)
+# Reading CSV pulled out into separate function.
 
-    all_blobs = list(client.list_blobs(bucket_name, prefix=prefix))
-    csv_path = next(filter(lambda blob: blob.name.endswith('.csv'), all_blobs)).name
 
-    blob = bucket.get_blob(csv_path).download_as_text()
-    csv_reader = csv.DictReader(io.StringIO(blob))
+def get_samples_from_db() -> List[str]:
+    """ Pulls list of samples to be moved from SampleMetadata DB """
 
+    sapi = SampleApi()
     samples: List[str] = []
-    for row in csv_reader:
-        samples.append(row['sample.sample_name'])
-
-    return samples, csv_path
 
+    # Returns a list of all samples with status == uploaded
+    # Alternatively, could return all samples that were not included in the latest .mt analysis.
+    samples = sapi.get_latest_uploads()  # TO IMPLEMENT: API CALL
 
-def determine_samples(
-    upload_bucket: str, upload_prefix: str, metadata_bucket: str, metadata_prefix: str
-) -> Tuple[Set[str], str]:
-    """Determine files that should be moved to main vs archive bucket.
-    Determines the difference between the latest CSV file within upload
-    and the CSV file from the most recent batch as the list of samples to
-    be moved.
-    """
-    all_samples: List[str] = []
-    previous_samples: List[str] = []
-
-    all_samples, csv_path_upload = samples_from_csv(upload_bucket, upload_prefix)
-    previous_samples, _ = samples_from_csv(metadata_bucket, metadata_prefix)
-
-    samples = set(all_samples) - set(previous_samples)
-
-    curr_csv_file_path = join(upload_bucket, csv_path_upload)
-
-    return samples, curr_csv_file_path
+    # The upload of these filse is to be handled in a separate function.
+    return samples
 
 
 def generate_file_list(
-    samples: Set[str],
+    samples: List[str],
```

Comment:
It would be worth clarifying this variable, eg: `internal_sample_ids` (and the subsequent `s` -> `internal_sample_id`)

Reply:
Have renamed the variables in this function to hopefully add clarity :D